src/libecalc/core/graph_result.py

```diff
@@ -295,6 +295,104 @@ def get_asset_result(self) -> libecalc.dto.result.EcalcModelResult:
                         unit=Unit.BARA,
                     ).for_period(period=period)
 
+                    rate = TimeSeriesRate(
+                        timesteps=model.timesteps,
+                        values=model.rate_sm3_day,
+                        unit=Unit.STANDARD_CUBIC_METER_PER_DAY,
+                        rate_type=RateType.STREAM_DAY,
+                    )
+
+                    # Convert rates in stage results from lists to time series:
+                    for stage_result in model.stage_results:
```

Comment:
Instead of manipulating the objects here, we should create a new ones. This way we don't end up with invalid data in model.stage_results, while also getting validation that the new objects are correct (pydantic validation).

When manipulating values pydantic does not validate, so here we get a situation where the type in the stage_result class says List[float] (or something) while the type has been altered to TimeSeriesRate.